### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by olligobber** [olligobber/DirectProof#1]
Expected behaviour of the program is to produce a counter-example or a proof. Instead it produced this error:

```
$ ./main "¬(¬(a=a)=(b=b))" "a->a" "b->b"
main: 
Failed to compose proofs, since the end of one did not match the
start of the other. The end of the first proof was
	Prop 1 :|: (Prop 2 :|: (((Prop 3 :|: (Prop 4 :|: Prop 5)) :|: (Prop 6 :|: Prop 7)) :|: ((Prop 8 :|: (Prop 9 :|: Prop 10)) :|: Prop 11)))
and the start of the second was
	(Not (Prop 12) :&: Prop 12) :|: ((Not (Prop 12) :&: (Prop 12 :&: (Not (Prop 13) :&: Prop 13))) :|: ((Not (Prop 12) :&: Not (Prop 13)) :|: ((Not (Prop 12) :&: Prop 13) :|: ((Prop 12 :&: Not (Prop 13)) :|: ((Prop 12 :&: Prop 13) :|: (Not (Prop 13) :&: Prop 13))))))
and when matching these, these two formulas could not be matched:
	(Prop 3 :|: (Prop 4 :|: Prop 5)) :|: (Prop 6 :|: Prop 7)
	Not (Prop 12) :&: Not (Prop 13)

CallStack (from HasCallStack):
  error, called at ./Proof.hs:80:20 in main:Proof
```

**Comment by olligobber** [comment]
Error seems to be in generating the proof between the DNF formulas `((¬a∧¬b)∨((¬a∧b)∨((a∧¬b)∨(a∧b))))` and `((¬a∧a)∨((¬a∧(a∧(¬b∧b)))∨((¬a∧¬b)∨((¬a∧b)∨((a∧¬b)∨((a∧b)∨(¬b∧b)))))))`.

**Comment by olligobber** [comment]
It seems to have trouble when the start has 3 or more clauses that are all in the goal.
```
$ ./main "a|(b|(c|d))" "a|(b|c)"
 1 |     (((t_1∨t_2)∨t_3)∨(b∨c))     | Assumption
 2 |     ((((t_1∨t_2)∨t_3)∨b)∨c)     | Association 1
 3 |     ((b∨((t_1∨t_2)∨t_3))∨c)     | Commutation 2
 4 |     (b∨(((t_1∨t_2)∨t_3)∨c))     | Association 3
 5 |     ((b∨((t_1∨t_2)∨t_3))∨c)     | Association 4
 6 |     (c∨(b∨((t_1∨t_2)∨t_3)))     | Commutation 5
 7 |     (c∨(((t_1∨t_2)∨t_3)∨b))     | Commutation 6
 8 |     ((c∨((t_1∨t_2)∨t_3))∨b)     | Association 7
 9 |     ((((t_1∨t_2)∨t_3)∨c)∨b)     | Commutation 8
10 |     (((t_1∨t_2)∨t_3)∨(c∨b))     | Association 9
11 |     (((t_1∨t_2)∨t_3)∨(b∨c))     | Commutation 10
12 | ((((t_1∨t_2)∨t_3)∨(b∨c))∨(c∨d)) | Addition 11
13 | (((t_1∨t_2)∨t_3)∨((b∨c)∨(c∨d))) | Association 12
14 | ((t_1∨(t_2∨t_3))∨((b∨c)∨(c∨d))) | Association 13
```

**Comment by olligobber** [comment]
Fixed in 32cda016f41ce1b5df96c2ec3fec0d33eb033c12 

Turns out I wrote Left instead of Right when lifting a proof
